simple-glue-transform/app.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+from typing import Any
+
+import aws_cdk as cdk
+import aws_cdk.aws_iam as iam
+from aws_cdk.aws_glue import CfnCrawler
+from aws_cdk.aws_glue_alpha import (Code, GlueVersion, JobExecutable,
+                                    JobLanguage, JobType)
+from aws_ddk_core.base import BaseStack
+from aws_ddk_core.pipelines import DataPipeline
+from aws_ddk_core.resources import S3Factory
+from aws_ddk_core.stages import GlueTransformStage, S3EventStage
+from constructs import Construct
+
+app = cdk.App()
+
+
+class DdkApplicationStack(BaseStack):
+    def __init__(
+        self, scope: Construct, id: str, environment_id: str, **kwargs: Any
+    ) -> None:
+        super().__init__(scope, id, environment_id, **kwargs)
+
+        bucket = S3Factory.bucket(
+            self,
+            id="bucket",
+            environment_id=environment_id,
+            event_bridge_enabled=True,
+        )
+
+        s3_stage = S3EventStage(
+            self,
+            id="s3-event-capture",
+            environment_id=self._environment_id,
+            event_names=["Object Created"],
+            bucket_name=bucket.bucket_name,
+            key_prefix="source",
+        )
+
+        glue_crawler_role = iam.Role(
+            self,
+            "glue-crawler-role",
+            assumed_by=iam.ServicePrincipal("glue.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSGlueServiceRole"
+                )
+            ],
+        )
+        glue_crawler_role.add_to_policy(
```

Comment:
I think it would be preferable to do
```python
bucket.grant_read(
    glue_crawler_role,
    objects_key_pattern="target/",  # optionally specify exactly which bucket prefix you need access to
)
```

Reply:
Agreed.